net/DescriptionViewController.m

```diff
@@ -86,104 +174,527 @@ -(void)issueImageDidLoad
+    [box.messageConstraintsBig addObject:[commentLabelMessage.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor constant:-60]];
+    
+    
+    
+    // Message button
+    buttonCommentMessage.translatesAutoresizingMaskIntoConstraints = NO;
+    buttonCommentMessage.restorationIdentifier = @"commentMessageButton";
+    [buttonCommentMessage addTarget:self action:@selector(commentMessageTapped:) forControlEvents:UIControlEventTouchUpInside];
+    buttonCommentMessage.tag = index;
+    box.buttonMessage = buttonCommentMessage;
     
+    [self.contentView addSubview:buttonCommentMessage];
+    box.lastZPos = buttonCommentMessage.layer.zPosition;
+    [buttonCommentMessage.topAnchor constraintEqualToAnchor:commentLabelMessage.topAnchor].active=YES;
+    [buttonCommentMessage.bottomAnchor constraintEqualToAnchor:commentLabelMessage.bottomAnchor].active=YES;
+    [buttonCommentMessage.leadingAnchor constraintEqualToAnchor:commentLabelMessage.leadingAnchor].active=YES;
+    [buttonCommentMessage.trailingAnchor constraintEqualToAnchor:commentLabelMessage.trailingAnchor].active=YES;
     
+    //commentView2 constrains
+    [commentView2.topAnchor constraintEqualToAnchor:avatar.topAnchor constant:-8].active = YES;
+    [commentView2.leadingAnchor constraintEqualToAnchor:avatar.leadingAnchor constant:-8].active = YES;
+    [commentView2.bottomAnchor constraintEqualToAnchor:commentLabelMessage.bottomAnchor constant:8].active = YES;
+    [commentView2.trailingAnchor constraintEqualToAnchor:commentLabelMessage.trailingAnchor constant:8].active = YES;
 }
 
+-(void)commentAvatarTapped:(UIButton*)sender
+{
+    [self callSegueToProfileWitTappedButton:sender];
+}
 
--(void)makeCategoryLabelWithStringCategory:(NSString*)stringCategory
+-(void)commentNameTapped:(UIButton*)sender
 {
-    NSString *firstPart = @"Category: ";
-    NSMutableAttributedString *aStr = [[NSMutableAttributedString alloc] initWithString:[firstPart stringByAppendingString:stringCategory]];
-    [aStr addAttribute:NSForegroundColorAttributeName value:[UIColor blueColor] range:NSMakeRange(firstPart.length, [aStr string].length - firstPart.length)];
-    dispatch_async(dispatch_get_main_queue(), ^{
-        self.currentCategoryLabel.attributedText = aStr;
-    });
+    [self callSegueToProfileWitTappedButton:sender];
+}
 
+-(void)callSegueToProfileWitTappedButton:(UIButton*) sender
+{
+    NSInteger index = sender.tag;
+    CommentBox *box = self.commentBoxArr[index];
+    self.callingSegueToProfileUserId = box.userID;
+    [self performSegueWithIdentifier:@"fromDescriptionToProfile" sender:self];
 }
 
+-(void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    if ([[segue identifier] isEqualToString:@"fromDescriptionToProfile"]) {
+        if([segue.destinationViewController isKindOfClass:[ProfileViewController class]])
+        {
+            ProfileViewController *profileViewController = (ProfileViewController*)segue.destinationViewController;
```

Comment:
This because of public API of profile controller and "harmony" in our command at all :(
Besides it userId - is not neessarily current user. On description viewcontroller user can segue to profile by tapping a name of commentator. (or avatar)